sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -387,6 +388,29 @@ def test_multi_task_lasso_and_enet():
     assert_true(0 < clf.dual_gap_ < 1e-5)
     assert_array_almost_equal(clf.coef_[0], clf.coef_[1])
 
+def test_lasso_readonly_data():
+    X = np.array([[-1], [0], [1]])
+    Y = np.array([-1, 0, 1])   # just a straight line
+    T = np.array([[2], [3], [4]])  # test sample
+
+    with TempMemMap((X, Y)) as (X, Y):
+        clf = Lasso(alpha=0.5)
+        clf.fit(X, Y)
+        pred = clf.predict(T)
+        assert_array_almost_equal(clf.coef_, [.25])
+        assert_array_almost_equal(pred, [0.5, 0.75, 1.])
+        assert_almost_equal(clf.dual_gap_, 0)
+
```

Comment:
style: PEP8 mandates 2 blank lines between top level code blocks (functions, classes) while 1 blank line to separate nested code blocks (e.g. methods).